### PR TITLE
Migrating non-auth components to TypeScript.

### PR DESCRIPTION
**Pull request by dennisoelkers**
## Motivation and Context

This PR is migrating the remaining, non-authentication components to TypeScript.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: graylog2-web-interface/src/components/common/IconButton.tsx [hunk 1]
```diff
@@ -49,7 +49,7 @@ type Props = {
   title: string,
   onClick?: () => void,
   className?: string,
-  name: string,
+  name?: string,
```
Comment (linuspahl) [inline graylog2-web-interface/src/components/common/IconButton.tsx:52]: Why did you typed this prop as optional?

File: graylog2-web-interface/src/components/graylog/Row.tsx [hunk 2]
```diff
@@ -33,7 +33,7 @@ export const RowContentStyles = css(({ theme }: RowContentType) => css`
   border-radius: 4px;
 `);
 
-const Row: StyledComponent<{children: any}, void, *> = styled(BootstrapRow)`
+const Row: StyledComponent<{children: any}, void, any> = styled(BootstrapRow)`
```
Comment (linuspahl) [inline graylog2-web-interface/src/components/graylog/Row.tsx:36]: We could replace `any` with `BootstrapRow`

**Review by linuspahl — COMMENTED** [review]
Looks good, just added one question regarding the `IconButton` change.

File: graylog2-web-interface/src/components/common/IconButton.tsx (hunk 1 above)
Comment (dennisoelkers) [reply graylog2-web-interface/src/components/common/IconButton.tsx]: It seemed to me as if it was not specified everytime the component is used, but it looks like I was wrong. Changed that!

**Review by linuspahl — APPROVED** [review]
Looks good to me!

**Pull request merged**